docs/ai/conceptual/prompt-engineering-in-dot-net.md

```diff
@@ -0,0 +1,190 @@
+Categories: Domestic Policy, US Economy, Foreign Affairs, Space Exploration, Other.</message>
+```
+
+### Azure OpenAI
+
+Azure OpenAI is a managed service that allows developers to deploy, tune, and generate content from OpenAI models on Azure resources. Azure OpenAI's client library for .NET is an adaptation of OpenAI's REST APIs that provides an idiomatic interface and rich integration with the rest of the Azure SDK ecosystem. It can connect to Azure OpenAI resources or to the non-Azure OpenAI inference endpoint, making it suitable for non-Azure OpenAI development.
+
+Use the client library for Azure OpenAI to:
+
+- Engineer generated text completions via the [Completion API](https://learn.microsoft.com/en-us/azure/ai-services/openai/concepts/advanced-prompt-engineering?pivots=programming-language-completions).
+- Engineer chat-based completions via the [Chat Completion API](https://learn.microsoft.com/en-us/azure/ai-services/openai/concepts/advanced-prompt-engineering?pivots=programming-language-chat-completions).
+
+Azure OpenAI Completions use GPT-3 and GPT-3.5 models, which have no specific format rules for prompts.
+
+```csharp
+Prompts = 
+    { 
+        "Summarize US Presidential accomplishments, grouped by category.",
+        "Categories: domestic policy, judicial appointments, trade agreements, space exploration", 
+    }
+```
+
+Azure OpenAI Chat Completions use GPT-35-Turbo and GPT-4 models, which use a specific chat-like format consisting of role-based messages.
+
+```csharp
+Messages =
+    {
+        // The system message represents instructions or other guidance about how the assistant should behave
+        new ChatRequestSystemMessage("You are helping students of US History with their homework. Be cheerful about it."),
+        // User messages represent current or historical input from the end user
+        new ChatRequestUserMessage("I need help with my homework."),
+        // Assistant messages represent historical responses from the assistant
+        new ChatRequestAssistantMessage("Of course! That's what I'm here for. How can I help?"),
+        new ChatRequestUserMessage("Summarize US Presidential accomplishments, grouped by category.\\n Categories: domestic policy, judicial appointments, trade agreements, space exploration"),
+    }
+```
+
+## Related content
+
+- [Prompt engineering techniques](https://learn.microsoft.com/en-us/azure/ai-services/openai/concepts/advanced-prompt-engineering)
```

Comment:
```suggestion
- [Prompt engineering techniques](/azure/ai-services/openai/concepts/advanced-prompt-engineering)
```

Reply:
I'd also consider switching this for a link to the _Chain of thought prompting_ doc

Reply:
Added a second related link targetting CoT prompting doc. GitHub prompted me to use a [def] instead of just a regular link target, so I did - might not have been the right choice but easy enough to revise.